File: example/boot.py


#encode/decode python code,bestter modify it yourself

def encode(ins:bytearray):
    '''
    :param ins:  bytearray bytecode
    :return: ins bytearray modifyed
    '''

    for i in range(len(ins)):
        ins[i] = ins[i]+2

    return ins


def decode(ins: bytearray):
    '''
    :param ins:  bytearray bytecode
    :return: ins bytearray modifyed
    '''

    for i in range(len(ins)):
        ins[i] = ins[i] - 2

    return ins

File: boot.c
//
// Created by Hasee on 2019/11/13.
//

#include "boot.h"
#include  "py_module_helper.h"
/*
 * boot_decode boot_encode for decode/encode boot python module
 * better modify it yourself
 * */

PyObject* boot_decode_impl(PyObject *ins){

    char *data = PyByteArray_AsString(ins);

    Py_ssize_t count = PyByteArray_GET_SIZE(ins);

    Py_ssize_t i =0;
    for(;i<count;i++)
        data[i]-=1;
    Py_XINCREF(ins);

    return ins;
}

PyObject* boot_encode_impl(PyObject *ins){

    char *data = PyByteArray_AsString(ins);

    Py_ssize_t count = PyByteArray_GET_SIZE(ins);

    Py_ssize_t i =0;
    for(;i<count;i++)
        data[i]+=1;

    Py_XINCREF(ins);

    return ins;
}